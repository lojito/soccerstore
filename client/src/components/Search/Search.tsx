import { useState, type FC } from "react";
import { styled } from "styled-components";

const DivWrapper = styled.div`
  width: 60%;
  display: inline-block;
  vertical-align: top;
  margin-top: 7px;
  position: relative;
  text-align: center;

  @media screen and (max-width: 950px) {
    width: 50%;
  }

  @media screen and (max-width: 720px) {
    width: 80%;
  }

  @media screen and (max-width: 330px) {
    width: 75%;
  }

  & input {
    margin: 0;
    width: 80%;
    padding: 8px 0 8px 6%;
    border-radius: 6px;
    border: 2px solid white;

    @media screen and (max-width: 950px) {
      padding-left: 10%;
    }

    @media screen and (max-width: 370px) {
      padding-left: 10%;
      width: 70%;
    }
  }

  & i {
    position: absolute;
    left: 8%;
    top: 9px;

    @media screen and (max-width: 370px) {
      left: 12%;
    }
  }
`;
type Props = {
  onSearch: (searchTerm: string) => void;
};

const Search: FC<Props> = ({ onSearch }) => {
  const [searchTerm, setSearchTerm] = useState("");

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.code === "Enter") {
      e.preventDefault();
      onSearch(searchTerm);
    }
  };

  return (
    <DivWrapper>
      <i className="fa fa-search" />
      <input
        type="text"
        id="text"
        name="search"
        placeholder="Buscar productos"
        value={searchTerm}
        onChange={(e) => setSearchTerm(e.target.value)}
        onKeyDown={(e) => handleKeyDown(e)}
      />
    </DivWrapper>
  );
};

export default Search;
