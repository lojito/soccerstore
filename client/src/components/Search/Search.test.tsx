import { render, screen } from "@testing-library/react";
import Search from "./Search";
import userEvent from "@testing-library/user-event";

describe("Search", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Search onSearch={vi.fn} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should allow to enter the search term and hit the enter key", async () => {
    const user = userEvent.setup();
    const searchTerm = "rojo";
    const onSearch = vi.fn();
    render(<Search onSearch={onSearch} />);

    const search = screen.getByRole("textbox");
    await user.type(search, searchTerm);
    await user.keyboard("[Enter]");

    expect(search).toHaveValue(searchTerm);
    expect(onSearch).toHaveBeenCalled();
  });
});
