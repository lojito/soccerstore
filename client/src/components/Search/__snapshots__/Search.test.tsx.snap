// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-beySPh iYHQvb"
  >
    <i
      class="fa fa-search"
    />
    <input
      id="text"
      name="search"
      placeholder="Buscar productos"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
